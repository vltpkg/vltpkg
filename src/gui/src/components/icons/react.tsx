export const React = (props: React.SVGProps<SVGSVGElement>) => {
  return (
    <svg
      width="500"
      height="500"
      viewBox="0 0 500 500"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}>
      <g clip-path="url(#clip0_2967_809)">
        <path
          d="m250.504 284.862c270.046 284.862 285.889 269.029 285.889 249.499c285.889 229.968 270.046 214.136 250.504 214.136c230.961 214.136 215.119 229.968 215.119 249.499c215.119 269.029 230.961 284.862 250.504 284.862z"
          fill="currentColor"
        />
        <path
          d="m250.501 321.96c355.363 321.96 440.371 289.522 440.371 249.509c440.371 209.495 355.363 177.058 250.501 177.058c145.639 177.058 60.6318 209.495 60.6318 249.509c60.6318 289.522 145.639 321.96 250.501 321.96z"
          stroke="currentColor"
          stroke-width="18.913"
        />
        <path
          d="m187.724 285.725c240.155 376.482 310.768 433.837 345.442 413.83c380.116 393.823 365.721 304.031 313.29 213.274c260.859 122.517 190.247 65.1621 155.572 85.1689c120.898 105.176 135.293 194.968 187.724 285.725z"
          stroke="currentColor"
          stroke-width="18.913"
        />
        <path
          d="m187.726 213.277c135.295 304.034 120.9 393.826 155.574 413.833c190.248 433.84 260.861 376.485 313.292 285.728c365.723 194.971 380.118 105.179 345.443 85.172c310.769 65.1652 240.157 122.52 187.726 213.277z"
          stroke="currentColor"
          stroke-width="18.913"
        />
      </g>
      <defs>
        <clipPath id="clip0_2967_809">
          <rect
            width="397"
            height="353"
            fill="currentColor"
            transform="translate(52 73)"
          />
        </clipPath>
      </defs>
    </svg>
  )
}
