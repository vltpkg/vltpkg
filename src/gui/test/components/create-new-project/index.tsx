import { vi, test, expect, afterEach } from 'vitest'
import { cleanup, render } from '@testing-library/react'
import html from 'diffable-html'
import { useGraphStore as useStore } from '@/state/index.js'
import { CreateNewProjectContent } from '@/components/create-new-project/index.jsx'

vi.mock('@/components/ui/card.jsx', () => ({
  Card: 'gui-card',
}))

vi.mock('@/components/ui/select.jsx', () => ({
  Select: 'gui-select',
  SelectContent: 'gui-select-content',
  SelectItem: 'gui-select-item',
  SelectTrigger: 'gui-select-trigger',
  SelectValue: 'gui-select-value',
}))

vi.mock('@/components/ui/input.jsx', () => ({
  Input: 'gui-input',
}))

vi.mock('@/components/ui/button.jsx', () => ({
  Button: 'gui-button',
}))

vi.mock('@/components/ui/form-label.jsx', () => ({
  Label: 'gui-label',
}))

expect.addSnapshotSerializer({
  serialize: v => html(v),
  test: () => true,
})

afterEach(() => {
  const CleanUp = () => (useStore(state => state.reset)(), '')
  render(<CleanUp />)
  cleanup()
})

test('create-new-project-content render default', async () => {
  render(<CreateNewProjectContent />)
  expect(window.document.body.innerHTML).toMatchSnapshot()
})

test('create-new-project-content with results', async () => {
  const Container = () => {
    const updateDashboard = useStore(state => state.updateDashboard)
    updateDashboard({
      buildVersion: '1.0.0',
      cwd: '/path/to/cwd',
      defaultAuthor: 'Ruy Adorno',
      dashboardProjectLocations: [
        { path: '/home/user', readablePath: '~' },
      ],
      projects: [
        {
          name: 'project-foo',
          readablePath: '~/project-foo',
          path: '/home/user/project-foo',
          manifest: { name: 'project-foo', version: '1.0.0' },
          tools: ['node', 'vlt'],
          mtime: 1730498483044,
        },
        {
          name: 'project-bar',
          readablePath: '~/project-foo',
          path: '/home/user/project-bar',
          manifest: { name: 'project-bar', version: '1.0.0' },
          tools: ['pnpm'],
          mtime: 1730498491029,
        },
      ],
    })
    return <CreateNewProjectContent />
  }
  render(<Container />)
  expect(window.document.body.innerHTML).toMatchSnapshot()
})
