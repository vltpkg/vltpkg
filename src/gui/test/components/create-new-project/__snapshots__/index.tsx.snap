// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`create-new-project-content render default 1`] = `

<div>
  <div class="flex grow flex-col bg-secondary px-8 py-8 dark:bg-black">
    <gui-card classname="container mx-auto w-full px-8 py-8 lg:w-[1024px]">
      <h1 class="mb-2 text-3xl font-bold">
        Create a new project
      </h1>
      <p class="mb-6 text-gray-600">
        A project is represented by a folder containing a package.json file.
      </p>
      <form class="space-y-6">
        <div class="flex space-x-4">
          <div class="w-1/2">
            <gui-label htmlfor="location">
              Location *
            </gui-label>
            <gui-select>
              <gui-select-trigger id="location">
                <gui-select-value placeholder="Select location">
                </gui-select-value>
              </gui-select-trigger>
              <gui-select-content>
              </gui-select-content>
            </gui-select>
          </div>
          <div class="w-1/2">
            <gui-label htmlfor="projectName">
              Project Name *
            </gui-label>
            <gui-input
              id="projectName"
              value
              classname
            >
            </gui-input>
          </div>
        </div>
        <div>
          <gui-label htmlfor="author">
            Author
          </gui-label>
          <gui-input
            id="author"
            value
          >
          </gui-input>
        </div>
        <div class="flex justify-end">
          <gui-button type="submit">
            Create New Project
          </gui-button>
        </div>
      </form>
    </gui-card>
  </div>
</div>

`;

exports[`create-new-project-content with results 1`] = `

<div>
  <div class="flex grow flex-col bg-secondary px-8 py-8 dark:bg-black">
    <gui-card classname="container mx-auto w-full px-8 py-8 lg:w-[1024px]">
      <h1 class="mb-2 text-3xl font-bold">
        Create a new project
      </h1>
      <p class="mb-6 text-gray-600">
        A project is represented by a folder containing a package.json file.
      </p>
      <form class="space-y-6">
        <div class="flex space-x-4">
          <div class="w-1/2">
            <gui-label htmlfor="location">
              Location *
            </gui-label>
            <gui-select>
              <gui-select-trigger id="location">
                <gui-select-value placeholder="Select location">
                </gui-select-value>
              </gui-select-trigger>
              <gui-select-content>
                <gui-select-item value="/home/user">
                  ~
                </gui-select-item>
              </gui-select-content>
            </gui-select>
          </div>
          <div class="w-1/2">
            <gui-label htmlfor="projectName">
              Project Name *
            </gui-label>
            <gui-input
              id="projectName"
              value
              classname
            >
            </gui-input>
          </div>
        </div>
        <div>
          <gui-label htmlfor="author">
            Author
          </gui-label>
          <gui-input
            id="author"
            value="Ruy Adorno"
          >
          </gui-input>
        </div>
        <div class="flex justify-end">
          <gui-button type="submit">
            Create New Project
          </gui-button>
        </div>
      </form>
    </gui-card>
  </div>
</div>

`;
